src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,67 @@
+package duke;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.LocalDate;
+
+/**
+ * Represents a <code>Task</code> with a deadline date
+ */
+public class Deadline extends Task {
+    private String deadlineDate;
+    private LocalDate date;
+
+    Deadline(String description, String deadlineDate) {
+        super(description);
+        this.deadlineDate = deadlineDate;
+        try {
+            new SimpleDateFormat("yyyy-mm-dd").parse(deadlineDate);
+            this.date = DateParser.parseDate(deadlineDate);
+        } catch (ParseException e) {
+            this.date = null;
+        }
+    }
+
+    Deadline(String description, String deadlineDate, Boolean completed) {
+        super(description, completed);
+        this.deadlineDate = deadlineDate;
+        try {
+            new SimpleDateFormat("yyyy-mm-dd").parse(deadlineDate);
+            this.date = DateParser.parseDate(deadlineDate);
+        } catch (ParseException e) {
+            this.date = null;
+        }
+    }
+
+    /**
+     * Obtains the type of <code>Task</code>
+     * @return String corresponding to the type of <code>Task</code>
+     */
+    @Override
+    public String getType() {
+        return "Deadline";
+    }
```

Comment:
Maybe can put task type as a class attribute
```suggestion
    // Class attribute
    static final private TYPE = "Deadline";
    
    @Override
    public String getType() {
        return TYPE;
    }
```